account_payment_purchase/models/purchase_order.py

```diff
@@ -7,19 +7,31 @@
 
 class PurchaseOrder(models.Model):
     _inherit = "purchase.order"
+    _check_company_auto = True  # missing in the native purchase module !!!
 
     supplier_partner_bank_id = fields.Many2one(
         comodel_name="res.partner.bank",
+        compute="_compute_payment_mode",
+        readonly=False,
+        store=True,
+        precompute=True,
         string="Supplier Bank Account",
-        domain="[('partner_id', '=', partner_id)]",
+        domain="[('partner_id', '=', partner_id), "
+        "'|', ('company_id', '=', company_id), ('company_id', '=', False)]",
```

Comment:
You can simplify the domain this way:

`[("company_id", "in", [False, company_id])]`

Reply:
done